pkg/stanza/fileconsumer/internal/archive/archive.go

```diff
@@ -0,0 +1,68 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package archive // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/archive"
+
+import (
+	"os"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/fileset"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/fingerprint"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/reader"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator"
+)
+
+const _ = "knownFiles"
+
+type Archive interface {
+	SetStorageClient(operator.Persister)
+	Match([]*FileRecord) ([]*reader.Reader, error)
+	Write([]*reader.Metadata) error
+}
+
+type FileRecord struct {
```

Comment:
Why do we need a different struct to store when we just store `reader.Metadata` everywhere else?

Reply:
different struct is needed because we also need the open file handle for reader creation later. Same reason as https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34415/#discussion_r1755164529

Reply:
Isn't the archive meant to be written to storage? It seems like an open file handle is not an option. Not to mention that we would be disregarding the max concurrency setting. 

Additionally, `knownFiles` doesn't even keep file handles, so why would we progress from `currentFiles` (open handles) -> `previousFiles` (open handles) -> `knownFiles` (no handles) -> `archive` (open handles again somehow?)

> Same reason as #34415

That's just a link to this PR.